src/installer.core.redscript.ts

```diff
@@ -0,0 +1,187 @@
+    ),
+  );
+
+const detectDeprecatedCoreRedscriptLayout = (fileTree: FileTree): boolean =>
+  pipe(
+    DEPRECATED_REDSCRIPT_CORE_REQUIRED_FILES,
+    RA.every(
+      (requiredFile) => pathInTree(requiredFile, fileTree),
+    ),
+  );
+
+const detectCoreRedscript = (fileTree: FileTree): boolean =>
+  detectCoreRedscriptLayout(fileTree) || detectDeprecatedCoreRedscriptLayout(fileTree);
+
+//
+// Layouts
+//
+
+const layout = (
+  _api: VortexApi,
+  _modName: string,
+  fileTree: FileTree,
+  layoutType: Layout,
+  layoutMatcher: (fileTree: FileTree) => boolean,
+): MaybeInstructions => {
+  //
+  if (!layoutMatcher(fileTree)) {
+    return NoInstructions.NoMatch;
+  }
+
+  const allProvidedFiles = filesUnder(FILETREE_ROOT, Glob.Any, fileTree);
+
+  const fileInstructions =
+    instructionsForSameSourceAndDestPaths(allProvidedFiles);
+
+  return {
+    kind: layoutType,
+    instructions: fileInstructions,
+  };
+};
+
+const coreRedscriptLayout = (
+  _api: VortexApi,
+  _modName: string,
+  fileTree: FileTree,
+): MaybeInstructions =>
+  layout(_api, _modName, fileTree, CoreRedscriptLayout.OnlyValid, detectCoreRedscriptLayout);
+
+const deprecatedCoreRedscriptLayout = (
+  _api: VortexApi,
+  _modName: string,
+  fileTree: FileTree,
+): MaybeInstructions =>
+  layout(_api, _modName, fileTree, CoreRedscriptLayout.Deprecated, detectDeprecatedCoreRedscriptLayout);
+
+//
+// Vortex API
+//
+
+// testSupported
+
+export const testForCoreRedscript: VortexWrappedTestSupportedFunc = (
+  api: VortexApi,
+  log: VortexLogFunc,
+  files: string[],
+  fileTree: FileTree,
+): Promise<VortexTestResult> =>
+  Promise.resolve({
+    supported:
+          detectCoreRedscript(fileTree),
+    requiredFiles: [],
+  });
+
+// install
+
+export const installCoreRedscript: VortexWrappedInstallFunc = async (
+  api: VortexApi,
+  _log: VortexLogFunc,
+  _files: string[],
+  fileTree: FileTree,
+  _destinationPath: string,
+  _progressDelegate: VortexProgressDelegate,
+) => {
+  //
+  const me = InstallerType.CoreRedscript;
+
+  const allPossibleCoreRedscriptLayouts = [
+    coreRedscriptLayout,
+    deprecatedCoreRedscriptLayout,
+  ];
+
+  const selectedInstructions = useFirstMatchingLayoutForInstructions(
+    api,
+    undefined,
+    fileTree,
+    allPossibleCoreRedscriptLayouts,
+  );
+
+  if (
+    selectedInstructions === NoInstructions.NoMatch ||
+    selectedInstructions === InvalidLayout.Conflict
+  ) {
+    //
+    const errorMessage = `Didn't Find Expected Core Redscript Installation!`;
+    api.log(
+      `error`,
+      `${me}: ${errorMessage}`,
+      sourcePaths(fileTree),
+    );
+
+    showWarningForUnrecoverableStructureError(
+      api,
+      me,
+      errorMessage,
+      sourcePaths(fileTree),
+    );
+    return Promise.reject(errorMessage);
+  }
+
+  // This would work better as a pipeline. Needs refactoring.
+  // `generateInstructions
+  //   |> checkForProblems
+  //   |> checkDeprecated
+  //   |> finalizeInstructions
+  //  `
```

Comment:
Just please give me elm or elixir pipes :( They look so niceee. And some pattern match along with it ;)